rackcity/views/user_views.py

```diff
@@ -16,29 +22,25 @@
     Status,
     get_user_permission_success,
 )
-from rackcity.utils.user_utils import is_netid_user
 from rackcity.utils.log_utils import (
     log_delete,
     log_user_permission_action,
     ElementType,
     PermissionAction,
 )
+from rackcity.utils.user_utils import is_netid_user
 import requests
+from rest_auth.registration.views import RegisterView
 from rest_framework.authtoken.models import Token
 from rest_framework.decorators import permission_classes, api_view
 from rest_framework.parsers import JSONParser
-from rest_framework.permissions import IsAuthenticated, IsAdminUser
-from rest_auth.registration.views import RegisterView
-from rackcity.permissions.groups import (
-    GroupName,
-    update_user_groups,
-    update_user_datacenter_permissions,
-)
+from rest_framework.permissions import IsAuthenticated
 
 
-class RegisterNameView(RegisterView):
+class RegisterNameView(PermissionRequiredMixin, RegisterView):
     serializer_class = RegisterNameSerializer
-    permission_classes = [IsAdminUser]
+    permission_required = PermissionPath.MODEL_WRITE.value
+    raise_exception = True
```

Comment:
I'm not sure what this is but why does it require model write permission?

Reply:
NICE CATCH i copy pasted the wrong thing. you need USER_WRITE permission for this tehe. FYSA this is a wrapper around django-rest-auth's user registration endpoint that we're using. it's the only class based view in the whole world that ive ever written

Reply:
In the whole wide world ever?